psi4/src/psi4/libmints/mintshelper.cc

```diff
@@ -876,14 +878,23 @@ SharedMatrix MintsHelper::ao_eri(std::shared_ptr<IntegralFactory> input_factory)
         factory = integral_;
     }
 
-    return ao_helper("AO ERI Tensor", std::shared_ptr<TwoBodyAOInt>(factory->eri()));
+    std::vector<std::shared_ptr<TwoBodyAOInt>> ints_vec(nthread_);
+    ints_vec[0] = std::shared_ptr<TwoBodyAOInt>(factory->eri());
+    for (size_t i = 1; i < nthread_; i++) {
+        ints_vec[i] = std::shared_ptr<TwoBodyAOInt>(ints_vec.front()->clone());
+    }
+    return ao_helper("AO ERI Tensor", ints_vec);
 }
 
 SharedMatrix MintsHelper::ao_eri(std::shared_ptr<BasisSet> bs1, std::shared_ptr<BasisSet> bs2,
                                  std::shared_ptr<BasisSet> bs3, std::shared_ptr<BasisSet> bs4) {
     IntegralFactory intf(bs1, bs2, bs3, bs4);
-    std::shared_ptr<TwoBodyAOInt> ints(intf.eri());
-    return ao_helper("AO ERI Tensor", ints);
+    std::vector<std::shared_ptr<TwoBodyAOInt>> ints_vec(nthread_);
+    ints_vec[0] = std::shared_ptr<TwoBodyAOInt>(intf.eri());
+    for (size_t i = 1; i < nthread_; i++) {
+        ints_vec[i] = std::shared_ptr<TwoBodyAOInt>(ints_vec.front()->clone());
+    }
```

Comment:
This might actually be better to run in parallel. That way the memory can get allocated to the processor's local memory.